kubelet/datadog_checks/kubelet/kubelet.py

```diff
@@ -309,7 +309,16 @@ def _retrieve_stats(self):
         """
         Retrieve stats from kubelet.
         """
-        return self.perform_kubelet_query(self.stats_url).json()
+        try:
+            stats_response = self.perform_kubelet_query(self.stats_url)
+            if stats_response.status_code >= 200 and stats_response.status_code < 300:
+                return stats_response.json()
+            else:
```

Comment:
maybe you can add another check `elseif stats_response.status_code == 403:` in order to log a warning `connection refused on kubelet’s "/stats/summary"`

Reply:
The warning log below will fire for any non-OK status code. Not only 40x, but also 50x.
So, we should know what is going wrong in all cases.